fedimint-client/src/lib.rs

```diff
@@ -2358,6 +2361,10 @@ impl ClientBuilder {
     ///
     /// ## Panics
     /// If there was a primary module specified previously
+    #[deprecated(
+        since = "0.6.0",
+        note = "Use `with_primary_module_kind` instead, as the instance id can't be known upfront"
```

Comment:
Being able to select a specific primary module by id is theoretically useful. It's just not something that typically should be done. Maybe we should rename and document better, but ultimately keep?

Reply:
Deprecating probably doesn't help much as a separate step anyway.

Reply:
I'd say let's create a fn with a new name that retains the old behavior, but is strongly discouraged. I'd like to phase out the old one to get rid of old and likely incorrect usage.

Reply:
Yes, we can just rugpull that name. :D